runtime/lua/vim/lsp/util.lua

```diff
@@ -544,6 +545,74 @@ end
 -- function M.glob_to_regex(glob)
 -- end
 
+--- Parses the raw glob into an |lpeg| pattern. LPeg patterns natively support the "this" or "that"
+--- alternative constructions described in the LSP spec that cannot be expressed in a standard Lua pattern.
+---
+---@param pattern string The raw glob pattern
+---@return vim.lpeg.Pattern? pattern An |lpeg| representation of the pattern, or nil if the pattern is invalid.
+function M.glob_to_lpeg(pattern)
```

Comment:
this change also would be helpful as a separate PR, then we can focus discussion for the workspace changes

Reply:
+1

And it probably shouldn't go into lsp.util, but another location. See https://github.com/neovim/neovim/issues/23291#issuecomment-1817873088 and https://github.com/neovim/neovim/issues/25272

And it would be nice if we could also document the supported glob pattern.

> Tests for vim.lsp.util.glob_to_lpeg

The `_watchfiles_spec` is currently mostly testing the pattern matching, we can move these too.